### PR TITLE
Undeprecate /_watcher endpoints

### PR DESCRIPTION
**Pull request by hub-cap**
We deprecated these awhile ago in favor of /_xpack/watcher
endpoints. Yet, this is no longer the direction that we want to go in,
instead we want to remove the /_xpack namespace. This commit
undeprecates the /_watcher endpoint.

### CODE REVIEW

File: x-pack/plugin/watcher/src/main/java/org/elasticsearch/xpack/watcher/rest/action/RestWatchServiceAction.java [hunk 1]
```diff
@@ -30,12 +30,17 @@ public RestWatchServiceAction(Settings settings, RestController controller) {
 
         // @deprecated Remove in 6.0
         // NOTE: we switched from PUT in 2.x to POST in 5.x
+        // NOTE: We added back the old URL with the new VERB (POST) since we are deprecating _xpack/* URIs in 7.0
```
Comment (hub-cap) [inline x-pack/plugin/watcher/src/main/java/org/elasticsearch/xpack/watcher/rest/action/RestWatchServiceAction.java]: im not 100% sure i did this right. Should I instead put both the PUT and POST back to non deprecated? In 7, only `POST /_watcher` and `POST /_xpack/watcher` exist, so this is why I did not undeprecate the `PUT /_watcher`

**Comment by hub-cap** [comment]
@elasticmachine run the gradle build tests 2

**Comment by hub-cap** [comment]
@elasticmachine run the gradle build tests 2

**Review by jasontedor — APPROVED** [review]
The change looks good, I left some comments about the comments.

**Pull request merged**